### PR TITLE
Delete Broadcast Functionality Added

### PR DESCRIPTION
**Pull request by amanjha21**
## Issue that this pull request solves

 Closes: #496

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

File: backend/app/routes/broadcast/deleteBroadcast.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const to = require('await-to-js').default;
+const Broadcast = require('../../models/Broadcast');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  const [err, broadcast] = await to(Broadcast.findByIdAndDelete(req.params.id));
+  if (!broadcast) {
+    res.status(400).send({
```
Comment (jackfrost13) [inline backend/app/routes/broadcast/deleteBroadcast.js:9]: Please use our error handler

File: backend/app/routes/broadcast/deleteBroadcast.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      message: "Broadcast doesn't exist",
+    });
+    return next();
+  }
+  if (err) {
+    const error = new ErrorHandler(constants.ERRORS.DATABASE, {
+      statusCode: 500,
+      message: 'Mongo Error: Insertion Failed',
```
Comment (jackfrost13) [inline backend/app/routes/broadcast/deleteBroadcast.js:17]: wrong message

File: backend/tests/routes/broadcast/deleteBroadcast.test.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        expect(res.body.isSuperAdmin).to.equal(true);
+        expect(res.body.token).to.be.a('string');
+        token = res.body.token;
+        done();
+      })
+      .catch(done);
+  });
+
+  // Step 2 - delete broadcast from DB
+  it('delete broadcast from DB at /broadcast/:id', (done) => {
```
Comment (jackfrost13) [inline backend/tests/routes/broadcast/deleteBroadcast.test.js:35]: would be good if you first add a broadcast then delete that

**Review by jackfrost13 — CHANGES_REQUESTED** [review]

File: backend/tests/routes/broadcast/deleteBroadcast.test.js (hunk 3 above)
Comment (amanjha21) [reply backend/tests/routes/broadcast/deleteBroadcast.test.js:35]: ok

File: backend/app/routes/broadcast/deleteBroadcast.js (hunk 1 above)
Comment (amanjha21) [reply backend/app/routes/broadcast/deleteBroadcast.js:9]: ok

File: backend/app/routes/broadcast/deleteBroadcast.js [hunk 4]
```diff
@@ -6,15 +6,16 @@ const constants = require('../../../constants');
 module.exports = async (req, res, next) => {
   const [err, broadcast] = await to(Broadcast.findByIdAndDelete(req.params.id));
   if (!broadcast) {
-    res.status(400).send({
+    const error = new ErrorHandler(constants.ERRORS.DATABASE, {
```
Comment (jackfrost13) [inline backend/app/routes/broadcast/deleteBroadcast.js]: This should be INPUT error

**Review by jackfrost13 — APPROVED** [review]

File: backend/app/routes/broadcast/deleteBroadcast.js (hunk 4 above)
Comment (amanjha21) [reply backend/app/routes/broadcast/deleteBroadcast.js]: ok

**Pull request merged**